src/runtime/async_read_ext.rs

```diff
@@ -0,0 +1,30 @@
+use async_trait::async_trait;
+use futures::io::AsyncRead;
+
+use crate::error::Result;
+
+/// Trait providing helpers that read various integer types in little-endian order.
+#[async_trait]
+pub(crate) trait AsyncLittleEndianRead: Unpin + futures::io::AsyncReadExt {
```

Comment:
One thing I've seen done before for traits that should only be implemented generically (like done below) is to make a private subtrait that this trait requires, e.g.

```rust
pub(crate) trait AsyncLittleEndianRead: Sealed + Unpin + futures::io::AsyncReadExt { ... }

trait Sealed {}

impl<R: AsyncRead + ?Sized + Unpin> Sealed for R {} 
```

This would prevent any outside implementations of `AsyncLittleEndianRead`. It's probably overkill given that this isn't a public trait, but I figured it was worth mentioning the pattern anyways.